plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -442,3 +442,32 @@ def find_dependent_rpms(self, units):
                 ret.add(unit)
 
         return ret
+
+    def find_dependent_rpms_relaxed(self, units):
+        pool = self.pool
+        done = set()
+        for unit in units:
+            solvable = self.mapping.get_solvable(unit)
+            if not solvable:
+                continue
+            candq = set([solvable])
+            while candq:
+                candq_n = set()
+                for cand in candq:
+                    _LOGGER.debug('Processing %s', cand)
+                    for req in cand.lookup_deparray(solv.SOLVABLE_REQUIRES):
+                        providers = pool.whatprovides(req)
+                        if not providers:
+                            _LOGGER.debug('Nothing provides %r', req)
+                            continue
+                        for p in pool.whatprovides(req):
```

Comment:
this is not conservative, it will coppy multiple providers or capability. imagine that you are pulling in package which has `Requires: system-release`. All `fedora-release-workstation`, `fedora-release-server`, `fedora-release-*` will be copied over =)

Reply:
yup it's the second mode of operation --- the follow-up that will make the relaxed way the default so that @gmbnomis doesn't suffer from the conservative change of behaviour.

Reply:
@dparalen I don't suppose that you are considering "relaxed" just for me, but to be sure: Don't do "relaxed" just for our use case. I think we can cope with "conservative" (Basically, I would ensure that the destination repo is "sane" by being empty. In a second step I could copy the rpms from this repo into the actual destination repo).

Reply:
@gmbnomis I agree that we should avoid surprising changes in updates esp. such that imply changing of workflows.
The conservative solver might be frustrating to use even with this patch.
I'd prefer users not frustrated ;)